### ISSUE TITLE
howto accumulate error when `cirterion:cuda()` is used

### ISSUE THREAD

**Issue by russellfei** [soumith/galaxyzoo#20]
hi, ~
currently, I'm learning `torch7` according to this repo, especially for implementation on GPU. And I'm confused how to `reduce` all errors calcualted by each kernel after `cirtierion:cuda()` is called. In this repo, this usage is commented.
Can anyone give me some advice or hints about `criterion:cuda()` ? thanks~

**Comment by soumith** [comment]
The training happens in 3_train.lua over here:
https://github.com/soumith/galaxyzoo/blob/127238921a339d157217662efd2019bceeae81f5/3_train.lua#L57
You calculate the gradients w.r.t. the criterion and then you use Stochastic Gradient Descent to go closer to your ideal kernel weights.
